proto/src/lib.rs

```diff
@@ -232,5 +232,202 @@ pub mod resource_recorder {
 }
 
 pub mod logger {
+    use std::time::Duration;
+
+    use prost::bytes::Bytes;
+    use tokio::{select, sync::mpsc, time::interval};
+    use tonic::{
+        async_trait,
+        codegen::{Body, StdError},
+        Request,
+    };
+    use tracing::error;
+
+    use self::logger_client::LoggerClient;
+
     include!("generated/logger.rs");
+
+    /// Adapter to some client which expects to receive a vector of items
+    #[async_trait]
+    pub trait VecReceiver: Send {
+        type Item;
+
+        async fn receive(&mut self, items: Vec<Self::Item>);
+    }
+
+    #[async_trait]
+    impl<T> VecReceiver for LoggerClient<T>
+    where
+        T: tonic::client::GrpcService<tonic::body::BoxBody> + Send + Sync + Clone,
+        T::Error: Into<StdError>,
+        T::ResponseBody: Body<Data = Bytes> + Send + 'static,
+        T::Future: Send,
+        <T::ResponseBody as Body>::Error: Into<StdError> + Send,
+    {
+        type Item = LogItem;
+
+        async fn receive(&mut self, items: Vec<Self::Item>) {
+            match self
+                .store_logs(Request::new(StoreLogsRequest { logs: items }))
+                .await
+            {
+                Ok(_) => {}
+                Err(error) => error!(
+                    error = &error as &dyn std::error::Error,
+                    "failed to send batch logs to logger"
+                ),
+            }
```

Comment:
We can use `if let Err(error) = ...` if we are not going to handle the `Ok(_)` case here.